src/transformers/models/whisper/generation_whisper.py

```diff
@@ -1131,33 +1135,52 @@ def replace_or_add(lst: List[int], num: int, itr: Iterator[int]):
                     f"You are using token ids in `forced_decoder_ids` that do not seem to correctly follow the prompt pattern of Whisper. Make sure that {forced_decoder_ids} has an entry for all indices >= 1 and < {forced_decoder_ids[0][0]}.",
                 )
 
-        # from v4.39 the forced decoder ids are always None in favour of decoder input ids
-        generation_config.forced_decoder_ids = None
-
         is_lang_id_undefined = len(init_tokens) <= 1 or (len(init_tokens) > 1 and init_tokens[1] is None)
-        if language is not None:
-            if language in generation_config.lang_to_id.keys():
-                language_token = language
-            elif language in TO_LANGUAGE_CODE.keys():
-                language_token = f"<|{TO_LANGUAGE_CODE[language]}|>"
-            elif language in TO_LANGUAGE_CODE.values():
-                language_token = f"<|{language}|>"
-            else:
-                is_language_code = len(language) == 2
-                raise ValueError(
-                    f"Unsupported language: {language}. Language should be one of:"
-                    f" {list(TO_LANGUAGE_CODE.values()) if is_language_code else list(TO_LANGUAGE_CODE.keys())}."
+
+        # Make sure language is a list of strings of the correct length
+        if isinstance(language, (list, tuple)):
+            if any(l is None for l in language):
+                raise TypeError(
+                    "Expected `language` to be `None`, a single string or a list of strings. Got a list containing `None`"
                 )
-            if language_token not in generation_config.lang_to_id:
+            if len(language) != batch_size:
                 raise ValueError(
-                    f"{language_token} is not supported by this specific model as it is not in the `generation_config.lang_to_id`."
-                    "(You should just add it to the generation config)"
+                    "When passing a list of languages, the length of the list must match the batch size. "
+                    f"Expected length of {batch_size}, but got {len(language)}"
                 )
+        else:
+            language = [language]  # Use a length-1 list now, broadcast later
+
+        # Separate init_tokens for each language
+        init_tokens = [copy.deepcopy(init_tokens) for _ in language]
+
+        # from v4.39 the forced decoder ids are always None in favour of decoder input ids
```

Comment:
Ah ok it's added back here. Could we move it back up so it sits next to the forced decoder id logic?